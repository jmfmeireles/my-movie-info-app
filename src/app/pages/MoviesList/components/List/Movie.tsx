import React from 'react';

import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';

import { MovieMetadata } from '../../types';
import {
  MovieWrapper,
  Panel,
  Name,
  MovieImage,
  Crew,
  YearAndRating,
  Year,
  RatingWrapper,
  Rating,
} from './style';

export default function Movie(props: { movie: MovieMetadata }) {
  const { t } = useTranslation();
  const history = useHistory();
  return (
    <MovieWrapper onClick={() => history.push(`/movie/${props.movie.id}`)}>
      <Panel>
        <MovieImage>
          <img src={props.movie.image} alt={props.movie.title} />
        </MovieImage>
        <Name>{props.movie.title}</Name>
        <Crew>{t('moviesListPage.crew', { crew: props.movie.crew })}</Crew>
        <YearAndRating>
          <Year>{props.movie.year}</Year>
          <RatingWrapper>
            <Rating />
            <p>{props.movie.imDbRating}</p>
          </RatingWrapper>
        </YearAndRating>
      </Panel>
    </MovieWrapper>
  );
}
