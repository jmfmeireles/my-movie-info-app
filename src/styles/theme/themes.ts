import { createTheme } from '@mui/material/styles';

const lightTheme = {
  primary: '#EDBF17',
  textPrimary: '#666666',
  textSecondary: 'rgba(58,52,51,0.7)',
  header: '#ffffff',
  footer: '#ffffff',
  background: '#F8F8F8',
  backgroundVariant: 'rgba(251,249,249,1)',
  border: 'rgba(58,52,51,0.12)',
  borderLight: 'rgba(58,52,51,0.05)',
};

export type Theme = typeof lightTheme;

export const themes = {
  light: lightTheme,
};

export const materialUiTheme = createTheme({
  typography: {
    fontFamily: ['"Gruppo"', 'sans-serif'].join(','),
  },
  components: {
    MuiChip: {
      styleOverrides: {
        avatar: {
          backgroundColor: lightTheme.primary,
          color: '#fff',
        },
      },
    },
  },
  palette: {
    primary: { main: lightTheme.primary },
  },
});
