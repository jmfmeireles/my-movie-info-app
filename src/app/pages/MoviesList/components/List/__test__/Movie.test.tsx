import React from 'react';

import { configure, mount } from 'enzyme';
import { createMemoryHistory } from 'history';
import { fireEvent, render } from '@testing-library/react';
import { BrowserRouter, Router } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import Movie from '../Movie';
import { themes } from '../../../../../../styles/theme/themes';
import moviesListMock from '../../../resources/mocks/movies_list_mock.json';

describe('Movie', () => {
  configure({ adapter: new Adapter() });

  it('Renders page properly', () => {
    expect(
      mount(
        <ThemeProvider theme={themes.light}>
          <BrowserRouter>
            <Movie movie={moviesListMock[0]} />
          </BrowserRouter>
        </ThemeProvider>,
      ),
    ).toMatchSnapshot();
  });

  it('Navigates to Movie Details page when clicking on the movie panel', () => {
    const history = createMemoryHistory();
    // mock push function
    history.push = jest.fn();

    const { getByTestId } = render(
      <ThemeProvider theme={themes.light}>
        <Router history={history}>
          <Movie movie={moviesListMock[0]} />
        </Router>
      </ThemeProvider>,
    );

    const movieWrapper = getByTestId(`movieWrapper`);
    fireEvent.click(movieWrapper);
    expect(history.push).toHaveBeenCalledWith(`/movie/${moviesListMock[0].id}`);
  });
});
