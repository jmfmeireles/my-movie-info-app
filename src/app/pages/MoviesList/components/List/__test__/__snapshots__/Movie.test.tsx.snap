// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Renders page properly 1`] = `ReactWrapper {}`;
